odoo/api.py

```diff
@@ -544,9 +569,14 @@ def companies(self):
         """ return a recordset of the enabled companies by the user """
         try:  # In case the user tries to bidouille the url (eg: cids=1,foo,bar)
             allowed_company_ids = self.context.get('allowed_company_ids')
-            # Prevent the user to enable companies for which he doesn't have any access
-            users_company_ids = self.user.company_ids.ids
-            allowed_company_ids = [company_id for company_id in allowed_company_ids if company_id in users_company_ids]
+            if not self.su:
+                # Prevent the user to enable companies for which he doesn't have any access
+                user_company_ids = self.user.company_ids.ids
+                allowed_company_ids = [cid for cid in allowed_company_ids if cid in user_company_ids]
```

Comment:
if not allowed_company_ids (there is an allowed_company_ids ctxt key but no key in it is in the user companies), should we fallback on user.company_ids.ids also ??